modules/pipeline/pkg/container_provider/container_instance_provider.go

```diff
@@ -67,19 +68,29 @@ func WithLabels(labels map[string]string) Option {
 
 // WithStages if the stages contain custom-type action, then it will make a disabled container instance provider
 // todo judge the container instance type in task-level
-func WithStages(stages []*pipelineyml.Stage) Option {
+func WithStages(extSvc *extmarketsvc.ExtMarketSvc, stages []*pipelineyml.Stage) Option {
```

Comment:
I have to say this dependency is strange.

Reply:
adjusted